interfaces/builtin/gpio_control.go

```diff
@@ -44,6 +44,7 @@ const gpioControlConnectedPlugAppArmor = `
 
 /sys/class/gpio/{,un}export rw,
 /sys/class/gpio/gpio[0-9]*/{active_low,direction,value,edge} rw,
+/sys/devices/platform/*/gpio/gpio[0-9]*/{active_low,direction,value,edge} rw,
```

Comment:
But is not this covered already by the gpio interface? It should de-reference `/sys/class/gpio/gpio<number>` and allow access to it with snippet `<dereferenced_path>/* rwk,`:
https://github.com/snapcore/snapd/blob/master/interfaces/builtin/gpio.go#L83

Reply:
Hi, as the one who raised original bugreport and without deep understanding of snapd/apparmor I think the code you linked is only applied on GPIO interfaces for a single GPIO pin. Individial GPIO plugs currently work without issue, just the generic `gpio-control` doesn't.

As the comment in that bit of code says `Given AppArmor requires symlinks to be dereferenced` the AppArmor profile should contain dereferenced paths. As far as I can see on the device the generated profile only contains the two `/sys/class/gpio` paths.

Reply:
Right, thanks for the explanation. In some cases you might need more control on export/unexport or gpios can appear dynamically in the system.